File: Algorithm/palindrome_number.py
class Solution:
    def isPalindrome(self,x):
        if x<0:
            return False
        div=1
        while x/div:
            div*=10
        div/=10
        while x and div:
            print x,div
            l=x/div
            r=x%10
            print l,r
            if l!=r:
                return False
            
            x-=l*div+r
            x/=10
            div/=100
        return True
if __name__=='__main__':
    sol=Solution()
    print sol.isPalindrome(2023883202)
                

File: Algorithm/CPP/palindrome_number.cpp
# include <iostream>
# include <string>
# include <stdlib.h>
# include <algorithm>
using namespace std;

class Solution {
public:
	bool isPalindrome(int x) 
	{
		if (x < 0 || (x != 0 && x % 10 == 0)) return false;
		int left = x;
		int right = 0;
		while (left > right)
		{
			right = right * 10 + left % 10;
			left = left / 10;
		}
		cout << left << " left" << endl;
		cout << right << " right" << endl; 
		return (left == right || left == right/10);

	}
};
int  main()
{
	Solution sol;

	//nums = { -1 };
	cout << sol.isPalindrome(0);
	cin.get();
	return 0;
}
